codec-http/src/main/java/io/netty/handler/codec/http/HttpResponseEncoder.java

```diff
@@ -28,6 +28,10 @@ public class HttpResponseEncoder extends HttpObjectEncoder<HttpResponse> {
 
     @Override
     public boolean acceptOutboundMessage(Object msg) throws Exception {
+        // common fast path
+        if (msg.getClass() == DefaultFullHttpResponse.class) {
```

Comment:
@normanmaurer 
This is rather uncomplete: `HttpResponseEncoder` should check against many concrete types that can be produced by users (almost all default http types, including content(s)), but this one is a widely adopted one, hence should be enough.

Users not satisfied can still override this method and provide the checks based on the types they believe are more common among their users (including custom types). wdyt?

Reply:
@slandelle I haven't implemented many of the HTTP optimizations for clients, but should be trivial to add them and benchmark them based on micros similar to the one I'm attaching to this PR; feel free to open a PR to fix it

Reply:
Can you extend the comment a bit to include the reason you outlined here ?